fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/implementation/PreAggregateTurnOnRule.java

```diff
@@ -323,12 +323,12 @@ public Void visitPhysicalHashJoin(OptExpression optExpression, PreAggregationCon
             List<ScalarOperator> rightGroupOperator = Lists.newArrayList();
 
             context.groupings.forEach(g -> {
-                if (leftOutputColumns.contains(g.getUsedColumns())) {
+                if (g.getUsedColumns().isIntersect(leftOutputColumns)) {
                     leftGroupOperator.add(g);
                 }
             });
             context.groupings.forEach(g -> {
-                if (rightOutputColumns.contains(g.getUsedColumns())) {
```

Comment:
The method name looks is inaccurate, can you add a new issue to modify it later?